hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/test/java/org/apache/hadoop/yarn/server/nodemanager/containermanager/linux/resources/TestCGroupsV2HandlerImpl.java

```
@@ -0,0 +1,287 @@
+        throw new IOException(message);
+      }
+    }
+    FileWriter mtabWriter = new FileWriter(mockMtab.getAbsoluteFile());
+    mtabWriter.write(baseCgroup2Line);
+    mtabWriter.close();
+    mockMtab.deleteOnExit();
+
+    String enabledControllers = "cpuset cpu io memory hugetlb pids rdma misc\n";
+
+    File controllersFile = new File(parentDir, CGroupsHandler.CGROUP_CONTROLLERS_FILE);
+    FileWriter controllerWriter = new FileWriter(controllersFile.getAbsoluteFile());
+    controllerWriter.write(enabledControllers);
+    controllerWriter.close();
+    controllersFile.deleteOnExit();
+
+    File subtreeControlFile = new File(parentDir, CGroupsHandler.CGROUP_SUBTREE_CONTROL_FILE);
+    Assert.assertTrue("empty subtree_control file should be created", subtreeControlFile.createNewFile());
+
+    File hierarchyDir = new File(parentDir, hierarchy);
+    if (!hierarchyDir.mkdirs()) {
+      String message = "Could not create directory " + hierarchyDir.getAbsolutePath();
+      throw new IOException(message);
+    }
+    hierarchyDir.deleteOnExit();
+    FileUtils.copyFile(controllersFile, new File(hierarchyDir, CGroupsHandler.CGROUP_CONTROLLERS_FILE));
+    FileUtils.copyFile(subtreeControlFile, new File(hierarchyDir, CGroupsHandler.CGROUP_SUBTREE_CONTROL_FILE));
+
+    return mockMtab;
+  }
+
+  @Test
+  public void testCGroupPaths() throws IOException {
+    verifyZeroInteractions(privilegedOperationExecutorMock);
+    CGroupsHandler cGroupsHandler = null;
+    File parentDir = new File(tmpPath);
+    File mtab = createPremountedCgroups(parentDir);
+    assertTrue("Sample subsystem should be created",
+        new File(controllerPath).exists());
+
+    try {
+      cGroupsHandler = new CGroupsV2HandlerImpl(createNoMountConfiguration(hierarchy),
+              privilegedOperationExecutorMock, mtab.getAbsolutePath());
+      cGroupsHandler.initializeCGroupController(controller);
+    } catch (ResourceHandlerException e) {
```

No need to catch the exception, let it throw and the we will see the whole stactrace, better for debug and less code.

---

Done.